### PR TITLE
[OpenSSL] Rename Debug libraries

### PR DESCRIPTION
**Pull request by Neumann-A**
reason see issue #6394

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @Neumann-A, thanks for your new PR. Here are some regressions from the current CI system:

|x64-windows|master| test |notes
|-----------|------|------|-----
|paho-mqtt  |Pass  |Fail  |**Regression**
|freerdp    |Pass  |Fail  |**Regression**
|cppcms     |Pass  |Fail  |**Regression**
|wt         |Pass  |Fail  |**Regression**

|x64-windows-static|master| test |notes
|------------------|------|------|-----
|paho-mqtt         |Pass  |Fail  |**Regression**
|freerdp           |Pass  |Fail  |**Regression**
|apr-util          |Pass  |Fail  |**Regression**

|arm64-windows|master| test |notes
|-------------|------|------|-----
|paho-mqtt    |Pass  |Fail  |**Regression**

|x86-windows|master| test |notes
|-----------|------|------|-----
|paho-mqtt  |Pass  |Fail  |**Regression**
|freerdp    |Pass  |Fail  |**Regression**
|cppcms     |Pass  |Fail  |**Regression**
|wt         |Pass  |Fail  |**Regression**

[failureLogs.zip](https://github.com/microsoft/vcpkg/files/3170985/failureLogs.zip)

**Comment by Neumann-A** [comment]
@NancyLi1013: There is something wrong with the logs here. I see the following line: 
`-- VCPKG-find_library-vars:OPENSSL_LIB;NAMES;ssl;NAMES_PER_DIR;libssl;ssleay32;HINTS;C:/OpenSSL-Win64/lib;C:/OpenSSL-Win64/lib64!`
which shouldn't be in this PR but is part of #5543

**Comment by NancyLi1013** [comment]
The new regressions:

|x64-windows|master| test |notes
|-----------|------|------|-----
|wt         |Pass  |Fail  |**Regression**

|x86-windows|master| test |notes
|-----------|------|------|-----
|wt         |Pass  |Fail  |**Regression**

[failureLogs.zip](https://github.com/microsoft/vcpkg/files/3189679/failureLogs.zip)

**Comment by Neumann-A** [comment]
Interesting regression:
```
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\ws2def.h(103): warning C4005: 'AF_IPX': macro redefinition
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\winsock.h(457): note: see previous definition of 'AF_IPX'
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\ws2def.h(140): warning C4005: 'AF_MAX': macro redefinition
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\winsock.h(476): note: see previous definition of 'AF_MAX'
```

means windows.h is included before winsock2.h. I am not even touching includes and CMake config shows no sign of doing something different......

**Comment by Neumann-A** [comment]
Build commands:

Fails (this PR):
`[383/415] C:\PROGRA~2\MIB055~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x86\cl.exe   /TP -DBOOST_ALL_DYN_LINK -DBOOST_SPIRIT_THREADSAFE -DWTISAPI_STATIC -DWT_BUILDING -DWT_FONTSUPPORT_DIRECTWRITE -DWT_WITH_OLD_INTERNALPATH_API -D_CRT_SECURE_NO_WARNINGS -D_REENTRANT -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0600 -I. -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-419193c3b7\src\web -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-419193c3b7\src -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-419193c3b7\src\Wt\Date\include -Isrc -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-419193c3b7\src\Wt\Dbo\backend\amalgamation -IG:\vcpkg_test\tmp\installed\x86-windows\include /DWIN32 /D_WINDOWS /W3  /GR /EHsc /MP  /bigobj /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    /MP /wd4267 /wd4251 /wd4275 /showIncludes /Fosrc\isapi\CMakeFiles\wtisapi.dir\IsapiRequest.C.obj /Fdsrc\isapi\wtisapid.pdb /FS -c G:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-419193c3b7\src\isapi\IsapiRequest.C`

Works (Master):
`[255/415] C:\PROGRA~2\MIB055~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x86\cl.exe   /TP -DBOOST_ALL_DYN_LINK -DBOOST_SPIRIT_THREADSAFE -DWTISAPI_STATIC -DWT_BUILDING -DWT_FONTSUPPORT_DIRECTWRITE -DWT_WITH_OLD_INTERNALPATH_API -D_CRT_SECURE_NO_WARNINGS -D_REENTRANT -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0600 -I. -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-0d346e3a00\src\web -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-0d346e3a00\src -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-0d346e3a00\src\Wt\Date\include -Isrc -IG:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-0d346e3a00\src\Wt\Dbo\backend\amalgamation -IG:\vcpkg_test\tmp\installed\x86-windows\include /DWIN32 /D_WINDOWS /W3  /GR /EHsc /MP  /bigobj /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    /MP /wd4267 /wd4251 /wd4275 /showIncludes /Fosrc\isapi\CMakeFiles\wtisapi.dir\IsapiRequest.C.obj /Fdsrc\isapi\wtisapid.pdb /FS -c G:\vcpkg_test\tmp\buildtrees\wt\src\4.0.5-0d346e3a00\src\isapi\IsapiRequest.C`

**Comment by Neumann-A** [comment]
@emweb: Please fix src/isapi/IsapiRequest.h including httpext.h which includes windows.h. If winsock2.h is not loaded before it build errors are certain. Please have also a look at the SSL and SQlite patches within this PR.

**Comment by Neumann-A** [comment]
Status/Regressions of/in the CI build?

**Comment by cbezault** [comment]
|x64-windows|master| test |notes
|-----------|------|------|-----
|paho-mqtt  |Pass  |Fail  |**Regression**
|freerdp    |Pass  |Fail  |**Regression**

|x64-windows-static|master| test |notes
|------------------|------|------|-----
|paho-mqtt         |Pass  |Fail  |**Regression**
|freerdp           |Pass  |Fail  |**Regression**
|apr-util          |Pass  |Fail  |**Regression**

|arm64-windows|master| test |notes
|-------------|------|------|-----
|paho-mqtt    |Pass  |Fail  |**Regression**

|x86-windows|master| test |notes
|-----------|------|------|-----
|paho-mqtt  |Pass  |Fail  |**Regression**
|freerdp    |Pass  |Fail  |**Regression**

**Comment by Neumann-A** [comment]
@cbezault: are you sure that this regression are for this PR? I couldn't reproduce the x86-windows ones and the apr-util one (I did not try the rest because it seems like something is wrong). If possible could you attache the logs so that I can check them?

**Comment by cbezault** [comment]
Yes those are for this PR, but the build has some other things that look questionable, I'll just re-run.

**Comment by Neumann-A** [comment]
@cbezault: Still the same issue? Or are there other regressions?

File: ports/openssl-windows/portfile.cmake [hunk 1]
```diff
@@ -172,6 +172,26 @@ file(READ "${CURRENT_PACKAGES_DIR}/include/openssl/rand.h" _contents)
 string(REPLACE "#  include <windows.h>" "#ifndef _WINSOCKAPI_\n#define _WINSOCKAPI_\n#endif\n#  include <windows.h>" _contents "${_contents}")
 file(WRITE "${CURRENT_PACKAGES_DIR}/include/openssl/rand.h" "${_contents}")
 
+#Rename required due to CMakes FindOpenSSL Module. It searches for names with an additional "d"
```
Comment (Neumann-A) [inline ports/openssl-windows/portfile.cmake]: forbidden see #6553

**Comment by Neumann-A** [comment]
Requires probably a different approach to rename the libraries. Reason see #6553.
Although only renaming the lib is probably ok.

**Comment by Neumann-A** [comment]
after doing a bit of research renaming only the library is seemingly ok.

**Comment by cbezault** [comment]
It seems like the ci is still having some problems with this PR, I'll give it one more shot and then will investigate more deeply.

**Comment by Neumann-A** [comment]
@cbezault any update on the CI here?

**Comment by cbezault** [comment]
|x64-windows     |test|notes
|----------------|----|-----
|signalrclient   |Fail|**Regression**
|google-cloud-cpp|Skip|Missing port dependency
|grpc            |Fail|**Regression**
|freerdp         |Fail|**Regression**
|openmama        |Fail|**Regression**
|paho-mqtt       |Fail|**Regression**

|x64-windows-static|test|notes
|------------------|----|-----
|signalrclient     |Fail|**Regression**
|arrow             |Skip|Missing port dependency
|google-cloud-cpp  |Skip|Missing port dependency
|aliyun-oss-c-sdk  |Skip|Missing port dependency
|grpc              |Fail|**Regression**
|freerdp           |Fail|**Regression**
|thrift            |Fail|**Regression**
|apr-util          |Fail|**Regression**
|parquet           |Skip|Missing port dependency
|paho-mqtt         |Fail|**Regression**

|arm64-windows   |test|notes
|----------------|----|-----
|paho-mqtt       |Fail|**Regression**
|grpc            |Fail|**Regression**
|google-cloud-cpp|Skip|Missing port dependency

|x86-windows     |test|notes
|----------------|----|-----
|signalrclient   |Fail|**Regression**
|google-cloud-cpp|Skip|Missing port dependency
|grpc            |Fail|**Regression**
|freerdp         |Fail|**Regression**
|openmama        |Fail|**Regression**
|paho-mqtt       |Fail|**Regression**

**Comment by cbezault** [comment]
[logs_2698798.zip](https://github.com/microsoft/vcpkg/files/3213102/logs_2698798.zip)

**Comment by Neumann-A** [comment]
@cbezault hmm these logs also belong to #5543 and not this PR.

**Comment by cbezault** [comment]
This is really weird, @Rastaban any idea what's going on?

**Comment by Neumann-A** [comment]
logs show something like that:
```
2019-05-22T22:36:05.3728840Z Previous HEAD position was c3c71c3f Merge 950bf0dc8af21b16dcae8561151fa7316e5b2d47 into ef2d3ff6d0c3cfc5754a640d0929d25aa13b4e14
2019-05-22T22:36:05.3729646Z HEAD is now at 1cbd1dd9 Merge 866b1c7be525ea38b43d74b0168950092dfc21b7 into ef2d3ff6d0c3cfc5754a640d0929d25aa13b4e14
```

but https://github.com/microsoft/vcpkg/commit/c3c71c3f is the other PR

**Comment by Neumann-A** [comment]
hmm but according to the logs x64-windows-static should be correctly checked out. There is something really strange going on. And you can easily check if it is from #5543 because it adds a lot of debug messages to the config of the form `VCPKG-<cmakefunction>`

**Comment by Neumann-A** [comment]
@cbezault Could it be that CI is not handling changes to the toolchain vcpkg.cmake file (or the whole scripts buildsystems folder) correctly and does not clean changes to it correctly?

**Comment by ras0219-msft** [comment]
@Neumann-A  Yes, we're currently not tracking changes to the toolchain vcpkg.cmake file -- this will be fixed by #6036

**Comment by cbezault** [comment]
@Neumann-A, I'm trying to get up to speed with this PR and I want to confirm a couple things.

1. The OpenSSL build system does not rename debug libraries?
2. The OpenSSL module shipped with CMake expects the debug libraries to be renamed?

**Comment by Neumann-A** [comment]
> @Neumann-A, I'm trying to get up to speed with this PR and I want to confirm a couple things.
> 
> 1. The OpenSSL build system does not rename debug libraries?
> 2. The OpenSSL module shipped with CMake expects the debug libraries to be renamed?

1. Yes. Debug libraries have the same name as release libraries
2. Yes. The OpenSSL installer probably bundles release and debug libraries or the names are different on linux and CMake uses that. The code used in CMake is citied in #6394 if you dont want to look in your cmake modules folder yourself ;)

Maybe just some flag is missing in the configure scripts? I cannot check it because the only thing I know is CMake ;)

**Comment by ras0219-msft** [comment]
An alternative to renaming the binaries would be to add a `vcpkg-cmake-wrapper.cmake` which could post-hoc modify the targets and variables coming out of the built-in `FindOpenSSL` module.

**Comment by Neumann-A** [comment]
Hmm after reading the install doc for openssl and comparing it with what vcpkg does I came to the conclusion that maybe a --debug is missing in the debug configure step.

**Comment by Neumann-A** [comment]
So I downloaded a binary installer for OpenSSL from https://slproweb.com/products/Win32OpenSSL.html

It clearly installs the debug libraries with additional suffix but has no extra debug dll. Thats probably where the find module within CMake gets the extra suffix. 

image

**Comment by Neumann-A** [comment]
Some problems with CI? Does not lock like a regression to me.
```
error: failed to update mingw32 (unable to lock database)
error: failed to update mingw64 (unable to lock database)
error: failed to update msys (unable to lock database)
error: failed to synchronize all databases
```

**Comment by Neumann-A** [comment]
I think this one is outdated with the update to 1.1 so I close it

**Pull request closed without merging**